src/functions.rs

```diff
@@ -16,36 +16,44 @@ import numpy as np
 from . import processing
 from scipy.fftpack import dct
 import math*/
-use ndarray::{Array1, Array2};
+use ndarray::{Array1, Array2, ArrayView1, ArrayViewMut1, ArrayViewMut2};
 
 /**
  * converting from frequency to Mel scale.
     :param f: The frequency values(or a single frequency) in Hz.
     :returns: The mel scale values(or a single mel).
 */
 pub fn frequency_to_mel(f: f64) -> f64 {
-    1127 * (1 + f / 700.).log()
+    1127. * (1. + f / 700.).ln()
 }
 //Note: may want to try this crate https://github.com/SuperFluffy/rust-expm
 /**
  * converting from Mel scale to frequency.
     :param mel: The mel scale values(or a single mel).
     :returns: The frequency values(or a single frequency) in Hz.
 */
-pub fn mel_to_frequency(mel: Array2<f64>) -> Array2<f64> {
-    700 * (np.exp(mel / 1127.0) - 1)
+pub fn mel_to_frequency(mel: Array1<f64>) -> Array1<f64> {
+    mel.map(|v| 700. * ((v / 1127.0).exp() - 1.))
 }
 
-pub fn triangle(x: Array1<f64>, left: f32, middle: f32, right: f32) -> Array1<f64> {
-    let mut out = Array1::<f64>::zeros(x.shape());
-    out[x <= left] = 0;
-    out[x >= right] = 0;
+pub fn triangle(arr: ArrayViewMut1<f32>, x: Array1<f32>, left: f32, middle: f32, right: f32) {
```

Comment:
Hey, I'm curious why the ArrayViewMut? There was something you couldn't achieve otherwise?

Reply:
yeah, it was mainly they were initializing filterbank and then initializing a second  matrix to fill a certain slice, which was not used afterwards. I'm attempting to operate directly on slices of the array, to avoid initializing matrices in a loop

Reply:
That's a good reasoning.